Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs

```diff
@@ -230,20 +235,34 @@ endRide handle@ServiceHandle {..} rideId req = withLogTag ("rideId-" <> rideId.g
     notifyCompleteToBAP booking updRide newFareParams
   return APISuccess.Success
 
-recalculateFareForDistance :: (MonadThrow m, Log m, MonadTime m, MonadGuid m) => ServiceHandle m -> SRB.Booking -> DRide.Ride -> Meters -> m (Meters, Money, Maybe FareParameters)
-recalculateFareForDistance handle@ServiceHandle {..} booking ride recalcDistance = do
+getMinuteDifference :: UTCTime -> UTCTime -> Int
+getMinuteDifference startTime endTime =
+  let diff = diffUTCTime endTime startTime
+      diffInMinutes = round (diff / 60)
+   in diffInMinutes
+
+waitingBlockCalculator :: DTConf.TransporterConfig -> DRide.Ride -> DBooking.Booking -> UTCTime -> Int
+waitingBlockCalculator config ride booking now = do
+  let rideStartTime = fromMaybe now ride.tripStartTime
+  let estimatedDurationInMinutes = floor ((fromIntegral booking.estimatedDuration :: Double) / 60) :: Int
+  let differenceInRideDuration = getMinuteDifference now rideStartTime - estimatedDurationInMinutes
+  let chargableBlocks = if fromIntegral differenceInRideDuration - config.waitingBlockInMinuites > 0 then floor (((fromIntegral differenceInRideDuration - config.waitingBlockInMinuites) :: Double) / config.waitingBlockInMinuites) :: Int else 0
```

Comment:
config.waitingBlockInMinutes is misspelt.
Also, this can't be used in both places in this calculation.
The value to be subtracted is 5 min and the denominator in division is 15 min currently. We would need two configs.
`config.freeWaitingTime` and `config.waitingBlockInMinutes`

Reply:
what will be the charge for pickup and trip per waiting block?

Reply:
trip per waiting block is 5₹. Pickup waiting can be kept 0 for now.